From 174cb91d82d643f78d8ba2b9999312b9d7dea98c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michel=20D=C3=A4nzer?= <mdaenzer@redhat.com>
Date: Mon, 7 Sep 2020 18:40:34 +0200
Subject: [PATCH xserver 09/16] present/wnmd: Remove dead check from
 present_wnmd_check_flip

present_wnmd_toplvl_pixmap_window returns a window with the same window
pixmap, so the check could never fail.

Reviewed-by: Roman Gilg <subdiff@gmail.com>
(cherry picked from commit b6b1161fd7ac409156fb69439897bcabdeacf393)
---
 present/present_wnmd.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/present/present_wnmd.c b/present/present_wnmd.c
index 6a3d1035b..35ea12968 100644
--- a/present/present_wnmd.c
+++ b/present/present_wnmd.c
@@ -270,10 +270,6 @@ present_wnmd_check_flip(RRCrtcPtr           crtc,
     if (!screen_priv->wnmd_info->flip)
         return FALSE;
 
-    /* Can't flip redirected child windows */
-    if (screen->GetWindowPixmap(window) != screen->GetWindowPixmap(toplvl_window))
-        return FALSE;
-
     /* Source pixmap must align with window exactly */
     if (x_off || y_off)
         return FALSE;
-- 
2.28.0

